leapp/repository/__init__.py

```diff
@@ -172,10 +172,12 @@ def _load_libraries(self, path=None, mod=None, prefix='leapp.libraries.common'):
             if is_pkg:
                 self._load_libraries(path=(os.path.dirname(loaded.__file__),), mod=loaded, prefix=mod_full_name)
 
-    def _load_modules(self, modules):
-        directories = tuple(os.path.join(self._repo_dir, os.path.dirname(module)) for module in modules)
-        for importer, name, is_pkg in pkgutil.iter_modules(directories):
-            importer.find_module(name).load_module(name)
+    def _load_modules(self, modules, mod_name=None):
+        for module in modules:
+            directory = os.path.join(self._repo_dir, os.path.dirname(module))
+            prefix = 'leapp.%s.' % directory.split(os.path.sep)[-1]
```

Comment:
Don't even try to make it that magically, just get the prefix to use from the caller. There is only function where this is called from with 4 calls where this is relevant.

Reply:
I would just change the calls in `def load` to this:

`self._load_modules(self.tags)` => `self._load_modules(self.tags, prefix='leapp.tags.')`
`self._load_modules(self.topics)` => `self._load_modules(self.topics, prefix='leapp.topics.')`
`self._load_modules(self.models)` => `self._load_modules(self.models, prefix='leapp.models.')`
`self._load_modules(self. workflows)` => `self._load_modules(self.workflows, prefix='leapp.workflows.')`

Pass through the prefix and be done with it. That's IMHO safer in case we change something about the directory structure. The assumption that basename(dirname(module_path)) gives you the right type, isn't a valid assumption. it could be easily models/something/somemodel.py

Reply:
Makes sense, will fix that. 
But this will change the semantics of _load_modules a bit, now it will be impossible to pass a list of heterogeneous modules, as they will be registered as one type only. Probably this should be put in a docstring.